### PR TITLE
High level overview

### PR DESCRIPTION
**Pull request by obonaventure**
Here is an attempt to provide a high level overview that summarizes the main principles of multipath QUIC. This should be expanded as we improve the text to align it with the full extension.

### CODE REVIEW

**Review by yfmascgy — COMMENTED** [review]
This overview puts a lot of emphasis on the algorithm side. I understand those algorithms eventually are integral parts if we want to make multipath quic work. But I feel this is a bit out of the scope of this very first draft. Maybe we should revisit this later on.

File: draft-lmbdhk-quic-multipath.md [hunk 1]
```diff
@@ -132,6 +132,68 @@ Both identifiers have the same value, which is the sequence number of the connec
...
 
+# High-level overview {#overview}
+
+
+The multipath extensions to QUIC proposed in this document enable the simultaneous utilization of 
+different paths to exchange non-probing QUIC frames for a single connection. This contrasts with
+the base QUIC protocol {{QUIC-TRANSPORT}} that includes a connection migration mechanism that
+selects only one path to exchange such frames.
+
+A multipath QUIC connection starts with a QUIC handshake as a regular QUIC connection.
```
Comment (mirjak) [inline draft-lmbdhk-quic-multipath.md:143]:
```suggestion
A multipath QUIC connection starts with a QUIC handshake as a regular QUIC connection.
See further Section {{nego}}.
```

File: draft-lmbdhk-quic-multipath.md [hunk 2]
```diff
@@ -132,6 +132,68 @@ Both identifiers have the same value, which is the sequence number of the connec
...
+The peers use the enable_multipath transport parameter during the handshake to
+negotiate the utilization of the multipath capabilities.
+The active_connection_id_limit transport parameter bounds the maximum number of paths
+that can be used during a connection. A multipath QUIC connection is thus an established QUIC
+connection where the enable_multipath transport parameter
+has been successfully negotiated.
+
+To add a new path to an existing multipath QUIC connection, a client starts a path validation on
+the chosen path. In this version of the document, a QUIC server does not initiate the creation
```
Comment (mirjak) [inline draft-lmbdhk-quic-multipath.md:152]:
```suggestion
the chosen path, as further described in Section {{setup}}.
In this version of the document, a QUIC server does not initiate the creation
```

File: draft-lmbdhk-quic-multipath.md [hunk 3]
```diff
@@ -132,6 +132,68 @@ Both identifiers have the same value, which is the sequence number of the connec
...
+The multipath extensions to QUIC proposed in this document enable the simultaneous utilization of 
+different paths to exchange non-probing QUIC frames for a single connection. This contrasts with
+the base QUIC protocol {{QUIC-TRANSPORT}} that includes a connection migration mechanism that
+selects only one path to exchange such frames.
+
+A multipath QUIC connection starts with a QUIC handshake as a regular QUIC connection.
+The peers use the enable_multipath transport parameter during the handshake to
+negotiate the utilization of the multipath capabilities.
+The active_connection_id_limit transport parameter bounds the maximum number of paths
+that can be used during a connection. A multipath QUIC connection is thus an established QUIC
```
Comment (huitema) [inline draft-lmbdhk-quic-multipath.md:147]: The active_connection_id_limit transport parameter *binds* the maximum number of paths

Comment (huitema) [reply draft-lmbdhk-quic-multipath.md:147]: The active CID limits binds the number of simultaneous paths. Or in fact, the number of simultaneous number spaces. It does not bind the total number of paths. When a path is abandoned, the CID is retired, the peer provides a new CID, and that new CID can be used on a new path.

File: draft-lmbdhk-quic-multipath.md [hunk 4]
```diff
@@ -132,6 +132,68 @@ Both identifiers have the same value, which is the sequence number of the connec
...
+
+Multipath QUIC uses acknowledgments like regularly QUIC. More details about these acknowledgments
+are provided later. Multipath QUIC enables new techniques to cope with packet losses compared to QUIC.
+In QUIC, a lost packet can either be abandoned or retransmitted over the same path once it has been
+detected as lost. A Multipath QUIC implementation can use different strategies to cope with losses.
+A simple strategy is to retransmit a lost frame over the same path. Another strategy would be to
+simultaneously retransmit a lost frame over its original path and another one.
+A third strategy would be to use a dedicated path to send retransmissions. This document does not
+preclude a specific strategy that would eventually depend on the application needs.
+
```
Comment (huitema) [inline draft-lmbdhk-quic-multipath.md:194]: I agree with Mirja that this is a bit too detailed, probably too long. The path creation and packet scheduling functions are local functions defined by the implementation. We cannot really standardize what implementations do, so the best is to just have a mention that these functions exist, and leave it at that.

**Review by huitema — CHANGES_REQUESTED** [review]
Please make it shorter...

File: draft-lmbdhk-quic-multipath.md [hunk 5]
```diff
@@ -132,6 +132,68 @@ Both identifiers have the same value, which is the sequence number of the connec
...
+The multipath extensions to QUIC proposed in this document enable the simultaneous utilization of 
+different paths to exchange non-probing QUIC frames for a single connection. This contrasts with
+the base QUIC protocol {{QUIC-TRANSPORT}} that includes a connection migration mechanism that
+selects only one path to exchange such frames.
+
+A multipath QUIC connection starts with a QUIC handshake as a regular QUIC connection.
+The peers use the enable_multipath transport parameter during the handshake to
+negotiate the utilization of the multipath capabilities.
+The active_connection_id_limit transport parameter bounds the maximum number of paths
```
Comment (mirjak) [inline draft-lmbdhk-quic-multipath.md:146]:
```suggestion
The active_connection_id_limit transport parameter limits the maximum number of active paths
```

**Review by mirjak — CHANGES_REQUESTED** [review]

File: draft-lmbdhk-quic-multipath.md [hunk 6]
```diff
@@ -132,6 +132,68 @@ Both identifiers have the same value, which is the sequence number of the connec
...
+manage the different paths. The first algorithm is the path manager (PM). Within the bounds
+set by the active_connection_id_limit transport parameter, the PM starts and stops paths based on
+the requirements of its application. The PM uses QUIC's path migration mechanism to create new paths and
+controls the termination of paths using the idle timeout and the exchange of PATH_ABANDON frames.
+It also monitors the utilization of the connection identifiers. We expect that different
+applications will use different paths managers based on their specific requirements. Since paths are
+always created by the client, a simple path manager would be to create one path per active network
+interface on the client. Another possibility would be to only create paths on the N first network
+interfaces of the client. A third possibility would be to only
+create additional paths if the existing path becomes congested.
```
[the remark this answers is not in the document]
Comment (qdeconinck) [reply draft-lmbdhk-quic-multipath.md:170]: I agree. I will try to rephrase this.

File: draft-lmbdhk-quic-multipath.md [hunk 7]
```diff
@@ -132,6 +132,68 @@ Both identifiers have the same value, which is the sequence number of the connec
...
+path's MTU as different paths do not necessarily have the same MTU. A simple PS is a round-robin
+scheduler that checks the congestion window of a path before sending a QUIC packet.
+Another sample PS is an algorithm that selects the path whose congestion window is open and
+that has the smallest round-trip-time. This is the PS used by the Linux implementation of Multipath
+TCP. Note that a multipath QUIC PS will probably treat control and data packets differently. For
+example, a PS could favor low delay paths to send acknowledgments and higher bandwidth
+ones for data packets. A companion draft
+{{I-D.bonaventure-iccrg-schedulers}} provides several general-purpose multipath
+packet schedulers. 
+
```
[the remark this answers is not in the document]
Comment (qdeconinck) [reply draft-lmbdhk-quic-multipath.md:187]: With the shortened version, I think the above paragraph can be moved in another PR that could indeed be placed in the scheduling section.

**Review by qdeconinck — COMMENTED** [review]
While the first part of the overview looks good, the second one is probably too detailed. I made some suggestions to simplify it.

File: draft-lmbdhk-quic-multipath.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (mirjak) [reply draft-lmbdhk-quic-multipath.md:187]: @qdeconinck can you create a new PR?

**Comment by qdeconinck** [comment]
Followed up by #108.

**Pull request closed without merging**